src/Config/Converters/Utf8JsonReaderExtensions.cs

```diff
@@ -41,4 +42,115 @@ static internal class Utf8JsonReaderExtensions
 
         return JsonSerializer.Deserialize<string>(ref reader, options);
     }
+
+    /// <summary>
+    /// Replaces placeholders for environment variables in the given JSON string and returns the new JSON string.
+    /// </summary>
+    /// <param name="inputJson">The JSON string to process.</param>
+    /// <exception cref="JsonException">Thrown when the input is null or when an error occurs while processing the JSON.</exception>
+    public static string? ReplaceEnvVarsInJson(string? inputJson)
+    {
+        try
+        {
+            if (inputJson is null)
+            {
+                throw new JsonException("Input JSON string is null.");
+            }
+
+            // Load the JSON string
+            using JsonDocument doc = JsonDocument.Parse(inputJson);
+            using MemoryStream stream = new();
+            using Utf8JsonWriter writer = new(stream, new JsonWriterOptions { Indented = true });
+
+            // Replace environment variable placeholders
+            ReplaceEnvVarsAndWrite(doc.RootElement, writer);
+            writer.Flush();
+
+            // return the final JSON as a string
+            return Encoding.UTF8.GetString(stream.ToArray());
+        }
+        catch (Exception e)
+        {
+            throw new JsonException("Failed to replace environment variables in given JSON. " + e.Message);
+        }
+    }
+
+    /// <summary>
+    /// Replaces placeholders for environment variables in a JSON element and writes the result to a JSON writer.
+    /// </summary>
+    /// <param name="element">The JSON element to process.</param>
+    /// <param name="writer">The JSON writer to write the result to.</param>
+    private static void ReplaceEnvVarsAndWrite(JsonElement element, Utf8JsonWriter writer)
+    {
+        switch (element.ValueKind)
+        {
+            case JsonValueKind.Object:
+                writer.WriteStartObject();
+                foreach (JsonProperty property in element.EnumerateObject())
+                {
+                    writer.WritePropertyName(property.Name);
+                    // Recursively process each property of the object
+                    ReplaceEnvVarsAndWrite(property.Value, writer);
+                }
+
+                writer.WriteEndObject();
+                break;
+            case JsonValueKind.Array:
+                writer.WriteStartArray();
+                foreach (JsonElement item in element.EnumerateArray())
+                {
+                    // Recursively process each item of the array
+                    ReplaceEnvVarsAndWrite(item, writer);
+                }
+
+                writer.WriteEndArray();
+                break;
+            case JsonValueKind.String:
+                string? value = element.GetString();
+                if (value is not null && value.StartsWith("@env('") && value.EndsWith("')"))
```

Comment:
Is this assuming that the start of the string is the beginning of the `env('<match>')`?

What happens if the `env('<match>')` happens in the middle of the string?

For example, if the person had 

{ 
  "database-type": "env('db_type')env('db_lang')" 
}

where their environment variable are db_type=ms, and db_lang=sql

will this replace `env('db_lang')` with `sql`?

Reply:
If you look at the JsonStringConverter the `StringJsonConverterFactory` there is logic that handles all of our edge cases with regex match and replace. If you can't use that class explicitly, perhaps you can factor out the logic that is being used for parsing the env variables to be shared?